core/amber/src/main/scala/edu/uci/ics/amber/engine/common/AmberUtils.scala

```diff
@@ -7,9 +7,21 @@ import edu.uci.ics.amber.engine.architecture.messaginglayer.DeadLetterMonitorAct
 
 import java.io.{BufferedReader, InputStreamReader}
 import java.net.URL
+import scala.collection.mutable.ArrayBuffer
 
 object AmberUtils {
 
+  /**
+    * Used by Reshape for predicting load on a worker. This is
+    * the estimated load using the mean model.
+    */
+  def predictedWorkload(workloads: ArrayBuffer[Long]): Double = {
+    var mean: Double = 0
+    workloads.foreach(load => { mean = mean + load })
+    mean = mean / workloads.size
+    mean
+  }
+
```

Comment:
hmm it's an awkward place to place this function since it is designed specifically for skew mitigation, and also this implementation is basically equivalent to `workloads.sum/workloads.size`. please consider move it some where that is not this common.

Reply:
you are right. Moved to the skew detection handler